GetDKAN/dkan-tools
# Fix multiple errors and confusion in make, related commands

This PR makes some changes to make getting started with DKAN2 a little easier:

1. Improves docs for the make commands
2. Turns off `--frontend` by default; use `dktl make --frontend` if you want the frontend
3. Won't fail when trying to build frontend but frontend already there
3. Simplifies symlink code and makes sure to delete and re-make symlinks in all cases (originally would try to make the symlink without deleting, producing an error)
4. Improves logic for updating drush; now gives warning if Drush is < 9 after make, and doesn't try to run update automatically. Also, update will only run if current version < const `DRUSH_VERSION` set in the BasicCommands class. Since these functions depend on each other, the update command comes back into BasicCommands.

## QA Steps

1. Start a new D8 project and run `dktl init` and `dktl get` commands
2. Run `dktl help make` and confirm that documentation is clearer
3. Run `dktl make --frontend` and confirm that you get a warning about the drush version
4. Run `dktl updatedrush` and confirm a) you get a warning about running this dangerous command and b) the update work
5. Run `dktl updatedrush` again and confirm that it doesn't try to do anything
6. Run `dktl make` again and confirm that you don't get any errors
